test/test_datasets_utils.py

```diff
@@ -215,6 +217,26 @@ def test_verify_str_arg(self):
         pytest.raises(ValueError, utils.verify_str_arg, 0, ("a",), "arg")
         pytest.raises(ValueError, utils.verify_str_arg, "b", ("a",), "arg")
 
+    @pytest.mark.parametrize(
+        ("dtype", "actual_hex", "expected_hex"),
+        [
+            (torch.uint8, "01 23 45 67 89 AB CD EF", "01 23 45 67 89 AB CD EF"),
+            (torch.float16, "01 23 45 67 89 AB CD EF", "23 01 67 45 AB 89 EF CD"),
+            (torch.int32, "01 23 45 67 89 AB CD EF", "67 45 23 01 EF CD AB 89"),
+            (torch.float64, "01 23 45 67 89 AB CD EF", "EF CD AB 89 67 45 23 01"),
```

Comment:
Not critical, but I'd prefer creating the tensors objects directly here; one shouldn't need to understand `frombuffer` nor `fromhex` to understand this test.

Reply:
As discussed offline, the direct way, e.g. `torch.tensor([0x00112233], dtype=torch.float32)` is not possible for floating point inputs. Although written as hexadecimal, the value is an integer and so `torch` converts it to a `float` which garbles the bytes. The only way to actually preserve the bytes regardless of the dtype is by going through `torch.frombuffer`.